### PR TITLE
Fix Headers Type.

### PR DESCRIPTION
**Pull request by wangbwn**
Fix Headers Type.

### CODE REVIEW

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (Dreamgoing) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json:7012]: change this type to object.

**Review by ruowan — APPROVED** [review]

**Comment by ms-henglu** [comment]
Please get a breaking changes approval from Azure Breaking Changes Reviewers <azbreakchangereview@microsoft.com> and ARM can review the changes afterwards. Once you have the approval, please remove the "ARMChangesRequested" label from the PR to make the PR visible to the ARM reviewers.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json
Comment (ruowan) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json:843]: Pls also change the Avro dataSet

**Pull request closed without merging**
